src/node/rpc/endpoint.h

```diff
@@ -47,12 +55,18 @@ namespace ccf
        {ForwardingRequired::Always, "always"},
        {ForwardingRequired::Never, "never"}});
 
+    DECLARE_JSON_ENUM(
+      ExecuteOutsideConsensus,
+      {{ExecuteOutsideConsensus::Never, "never"},
+       {ExecuteOutsideConsensus::Locally, "locally"},
+       {ExecuteOutsideConsensus::Primary, "primary"}});
```

Comment:
I'm trying to understand why there's 3 cases here - I would have expected just `Always` and `Never` - and how they interact with the forwarding options. What does a backup do with a `Primary` transaction - does this override the `ForwardingRequired` property and always get forwarded?

It feels like the default for a "command" should be "always execute locally". Forwarding to the primary, then executing outside of consensus, feels like an attempt to get a service-level answer for something which is really node-local (if there was a service-safe answer, it would be in the KV).

Reply:
Discussed offline - I renamed things as discussed. There will be followup following this PR (as discussed)